src/locales/en/parsers/ENCasualDateParser.ts

```diff
@@ -15,25 +15,35 @@ export default class ENCasualDateParser extends AbstractParserWithWordBoundaryCh
     innerExtract(context: ParsingContext, match: RegExpMatchArray): ParsingComponents | ParsingResult {
         let targetDate = dayjs(context.refDate);
         const lowerText = match[0].toLowerCase();
-        const component = context.createParsingComponents();
+        let component = context.createParsingComponents();
 
         switch (lowerText) {
             case "now":
-                return references.now(context.reference);
+                component = references.now(context.reference);
+                component.addTag("ENCasualDateParser/extract/now");
+                break;
 
             case "today":
-                return references.today(context.reference);
+                component = references.today(context.reference);
+                component.addTag("ENCasualDateParser/extract/today");
+                break;
 
             case "yesterday":
-                return references.yesterday(context.reference);
+                component = references.yesterday(context.reference);
+                component.addTag("ENCasualDateParser/extract/yesterday");
```

Comment:
I think we need a generic tag for that use case as well: `casualReference/yesterday`. This way users can identify the casual reference regardless of the language (or having to loop over all of them).

Same for the other matches below: `casualReference/tomorrow`, `casualReference/tonight`, `casualReference/last_night`.

Perhaps we can add another tag to identify the parser: `parser/ENCasualDateParser`.

Reply:
Thanks for the feedback. I'm also considering if we should add the tag in `ENCasualDateParser` or the `casualReferences`.

Seem that concepts such as 'tomorrow' should be tagged by 'casualReference', so it works orthogonally to the locale-specific. 

Maybe we could also simply add in both places. I don't think adding tags is that costly.

Reply:
> Seem that concepts such as 'tomorrow' should be tagged by 'casualReference', so it works orthogonally to the locale-specific.

Agreed.

> Maybe we could also simply add in both places. I don't think adding tags is that costly.

I thought about that too, adding `ENCasualDateParser/yesterday` as well as `casualReference/yesterday`.

We would still need `parser/ENCasualDateParser` (or similar), though.

It seems to me that `casualReference/yesterday` and `parser/ENCasualDateParser` are enough to know that it's coming from the `ENCasualDateParser` for the casual reference `yesterday`.

But as you said, having several tags is no big deal, even if they are a little redundant. Might actually make the developer experience better.

Reply:
I ended up separately adding tags in casualReference (`casualReference/yesterday`, `casualReference/evening`, etc) and parsers (`parser/ENCasualDateParser`, `parser/ENCasualTimeParser` etc) as you suggested.

I'll add tags in other parsers and refiners later, but please feel free to send PR to add more tags if there are specific parsers or contexts you need to recognize.

Reply:
Sorry for the delay, I was on holidays.

This is looking great, I'll be updating the library this week or the next. I'll send a PR if needed but I think we have everything we need as it stands.

Thanks again 🙏